### PR TITLE
[Notifier] Rework/streamline bridges (5.1)

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 5.1
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | ---
| License       | MIT
| Doc PR        | ---

This PR

* add missing tests
* pull up scheme check (check scheme first and then for required options)

### CODE REVIEW

File: src/Symfony/Component/Notifier/Bridge/RocketChat/composer.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "autoload": {
         "psr-4": { "Symfony\\Component\\Notifier\\Bridge\\RocketChat\\": "" },
         "exclude-from-classmap": [
-            "/Tests/"
+            "/Notifier/Bridge/Mattermost/Tests/"
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Notifier/Bridge/RocketChat/composer.json]: ?

Comment (OskarStark) [reply src/Symfony/Component/Notifier/Bridge/RocketChat/composer.json:26]: We can yes, I would recommend to do this across all bridges.

But what is the difference?

Comment (OskarStark) [reply src/Symfony/Component/Notifier/Bridge/RocketChat/composer.json]: Sorry this looks wrong :-)

Comment (OskarStark) [reply src/Symfony/Component/Notifier/Bridge/RocketChat/composer.json]: Fixed

**Comment by derrabus** [comment]
`"symfony/notifier": "~5.1.0"` to make Travis happy.

**Comment by OskarStark** [comment]
> "symfony/notifier": "~5.1.0" to make Travis happy.

Done, thanks 👍

**Comment by jderusse** [comment]
Could you please merge this PR #39454 with #39455?

**Comment by OskarStark** [comment]
Just for my curiosity, why? ☺️

**Comment by jderusse** [comment]
when the PRs contain the same thing, it's easier to review:
- when something has changed, it's easier to check if it has been applied in all components
- can review everything once (no need to open PR individually one by one)

**Comment by OskarStark** [comment]
Done @jderusse 👍

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/Notifier/Bridge/Mattermost/MattermostTransportFactory.php [hunk 2]
```diff
@@ -26,16 +27,22 @@ final class MattermostTransportFactory extends AbstractTransportFactory
...
+
         $token = $this->getUser($dsn);
         $channel = $dsn->getOption('channel');
-        $host = $dsn->getHost();
-        $port = $dsn->getPort();
 
-        if ('mattermost' === $scheme) {
-            return (new MattermostTransport($token, $channel, $this->client, $this->dispatcher))->setHost($host)->setPort($port);
+        if (!$channel) {
+            throw new IncompleteDsnException('Missing channel.', $dsn->getOriginalDsn());
```
Comment (jderusse) [inline src/Symfony/Component/Notifier/Bridge/Mattermost/MattermostTransportFactory.php:39]: Could you please add a test for this case?

Comment (OskarStark) [reply src/Symfony/Component/Notifier/Bridge/Mattermost/MattermostTransportFactory.php:39]: Sure

Comment (OskarStark) [reply src/Symfony/Component/Notifier/Bridge/Mattermost/MattermostTransportFactory.php:39]: Done

File: src/Symfony/Component/Notifier/Bridge/Mattermost/Tests/MattermostTransportFactoryTest.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        $accessToken = 'testAccessToken';
+        $host = 'testHost';
+        $channel = 'testChannel';
+
+        $transport = $factory->create(Dsn::fromString(sprintf('mattermost://%s@%s/?channel=%s', $accessToken, $host, $channel)));
+
+        $this->assertSame(sprintf('mattermost://%s?channel=%s', $host, $channel), (string) $transport);
+    }
+
+    public function testCreateWithMissingOptionFromThrowsIncompleteDsnException()
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Mattermost/Tests/MattermostTransportFactoryTest.php]:
```suggestion
    public function testCreateWithMissingOptionChannelThrowsIncompleteDsnException()
```

**Comment by OskarStark** [comment]
This one is ready to merge from my side 👍

File: src/Symfony/Component/Notifier/Bridge/Firebase/FirebaseOptions.php [hunk 4]
```diff
@@ -22,12 +22,10 @@
  */
 abstract class FirebaseOptions implements MessageOptionsInterface
 {
-    /** @var string the recipient */
+    /** the recipient */
```
Comment (derrabus) [inline src/Symfony/Component/Notifier/Bridge/Firebase/FirebaseOptions.php]: I would remove that doc block completely, tbh.

Comment (OskarStark) [reply src/Symfony/Component/Notifier/Bridge/Firebase/FirebaseOptions.php]: Agreed

Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/Firebase/FirebaseOptions.php]:
```suggestion
```

**Review by jderusse — APPROVED** [review]

**Comment by derrabus** [comment]
Thank you Oskar.

**Comment by OskarStark** [comment]
Thank you for all the reviews 😍

**Pull request merged**